lib/reek/detector_repository.rb

```diff
@@ -35,31 +35,25 @@ def initialize(smell_types: self.class.smell_types,
                    configuration: {})
       @configuration = configuration
       @smell_types   = smell_types
-      @detectors     = smell_types.map { |klass| klass.new configuration_for(klass) }
     end
 
     def examine(context)
-      smell_detectors_for(context.type).flat_map do |detector|
-        detector.run_for(context)
+      smell_detectors_for(context.type).flat_map do |klass|
```

Comment:
At first glance, it doesn't look like `#smell_detectors_for` would return a two-dimensional array. Could this get away with just `#map`?